core/modules/installer/installer.browser.inc

```diff
@@ -0,0 +1,470 @@
+  return isset($release_data['releases'][$release_name]) ? $release_data['releases'][$release_name] : FALSE;
+}
+
+/**
+ * Gets the newly installed projects from the session.
+ */
+function installer_browser_get_installed_projects() {
+  $projects = array();
+
+  if (isset($_SESSION['installer_browser_installed_projects'])) {
+    foreach ($_SESSION['installer_browser_installed_projects'] as $project) {
+      if (is_array($project) AND !empty($project)) {
+        $projects[$project['name']] = $project;
+      }
+    }
+  }
+
+  return $projects;
+}
+
+/**
+ * Adds a project to the install queue.
+ */
+function installer_browser_install_queue_add($project) {
+  $_SESSION['installer_browser_install_list'][$project['name']] = $project;
+}
+
+/**
+ * Removes a project from the install queue.
+ */
+function installer_browser_install_queue_remove($project_name) {
+  if (isset($_SESSION['installer_browser_install_list'][$project_name])) {
+    unset($_SESSION['installer_browser_install_list'][$project_name]);
+  }
+}
+
+/**
+ * Clears the install queue.
+ */
+function installer_browser_install_queue_clear() {
+  $_SESSION['installer_browser_install_list'] = array();
+}
+
+/**
+ * Gets the currently queued releases from the session.
+ */
+function installer_browser_get_queued_releases() {
+  $releases = array();
+
+  if (isset($_SESSION['installer_browser_install_releases_list'])) {
+    foreach ($_SESSION['installer_browser_install_releases_list'] as $release_name => $project) {
+      if (is_array($project) AND !empty($project)) {
+        $releases[$release_name] = $project;
+      }
+    }
+  }
+
+  return $releases;
+}
+
+/**
+ * Fetches results from the server based on the parameters passed in.
+ *
+ * $filters should be an associative array with the following keys:
+ * array(
+ *   'version' => '1', // The Major Version of Backdrop that is running on the Client
+ *   'text' => 'views', // The text that was entered as the search query, or '' if none
+ *   'type' => 'module', // The type of project being searched
+ *   'page' => 3, // The zero-based page number
+ *   'items_per_page' => 12, // How many results are requested per page
+ * )
+ *
+ * The installer_browser_fetch_results($filters) call returns an array like this:
+ *
+ * array(
+ *   'total' = 5, // The total number of results found for the filters
+ *   'projects' => array( // An array of projects returned for this page request
+ *     'views' => array( // A project array keyed by the machine name
+ *       'type' => 'module', // The type of project this is. Can be 'module' or 'theme'
+ *       'title' => 'Views', // The title of the project
+ *       'name' => 'views', // The machine name of the project
+ *       'author' => 'merlinofchaos', // The author's name
+ *       'description' => "Long project description ...",
+ *       'image' => 'http://www.example.com/image.jpg', // Absolute url to the image, if any
+ *       'usage' => '542312', // How many Downloads the module has had
```

Comment:
`How many Downloads the module has had` => `Number of reported active installs.`